python/pyarrow/_hdfs.pyx

```diff
@@ -134,19 +147,23 @@ cdef class HdfsOptions:
 
 
 cdef class HadoopFileSystem(FileSystem):
-    """HDFS backed FileSystem implementation
-
-    Parameters
-    ----------
-    options: HdfsOptions, default None
-        Options for connecting to hdfs. If None is passed then attempts to
-        initialize the connection from AWS environment variables.
-    """
+    """HDFS backed FileSystem implementation"""
 
     cdef:
         CHadoopFileSystem* hdfs
 
     def __init__(self, options_or_uri=None):
+        """Create HadoopFileSystem
+
+        Parameters
+        ----------
+        options_or_uri: HdfsOptions or str, default None
```

Comment:
We typically put the init parameters in the class docstring above

Reply:
We mix it through the codebase. Shall we setup numpydoc?

Reply:
> Shall we setup numpydoc?

I don't fully understand what you mean here. We already use numpydoc for parsing the docstrings in sphinx. The validation of numpydoc I was speaking about checks docstrings, not where they are put.

Reply:
I mean the validation:

```
$ python -m numpydoc --validate pyarrow.field 
pyarrow.field:GL01:Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pyarrow.field:SS03:Summary does not end with a period
pyarrow.field:ES01:No extended summary found
pyarrow.field:PR02:Unknown parameters {'metadata', 'nullable', 'name', 'type'}
pyarrow.field:PR06:Parameter "name" type should use "str" instead of "string"
pyarrow.field:PR07:Parameter "name" has no description
pyarrow.field:PR07:Parameter "type" has no description
pyarrow.field:PR06:Parameter "nullable" type should use "bool" instead of "boolean"
pyarrow.field:PR07:Parameter "nullable" has no description
pyarrow.field:PR09:Parameter "metadata" description should finish with "."
pyarrow.field:SA01:See Also section not found
pyarrow.field:EX01:No examples section found
```

Reply:
Ah, cool. If we want to use, we will need to disable a bunch of checks first I think. 

But it seems that it does not recognize the cython signature, as it says: "Unknown parameters {'metadata', 'nullable', 'name', 'type'}". Hmm, I thought this worked in pandas

Reply:
Even if we ignore certain errors it would still help to maintain the docstrings.